cfme/automate/provisioning_dialogs.py

```diff
@@ -118,25 +120,14 @@ class form(ProvDiagForm):  # noqa
         reset = Button('Reset')
 
 
-class ProvisioningDialog(Updateable, Pretty, Navigatable):
-    HOST_PROVISION = 'Host Provision'
-    VM_MIGRATE = 'VM Migrate'
-    VM_PROVISION = 'VM Provision'
-    SYSTEM_PROVISION = 'Configured System Provision'
-    ALLOWED_TYPES = {HOST_PROVISION, VM_MIGRATE, VM_PROVISION, SYSTEM_PROVISION}
-
+@attr.s
+class ProvisioningDialog(Updateable, Pretty, BaseEntity):
     pretty_attrs = ['name', 'description', 'diag_type', 'content']
 
-    def __init__(self, diag_type, name=None, description=None, content=None, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
-        self.name = name
-        self.description = description
-        self.content = content
-        if diag_type in self.ALLOWED_TYPES:
```

Comment:
**REQUIRED:** I do not believe we should use this validation.   It is happening in the right place (within the class that contains the attribute, and it provides useful information if someone used something other than one of these ALLOWED_TYPES.   This shows that you can define a validator for an attribute, so you might be able to use that somehow:   http://www.attrs.org/en/stable/api.html#attr.Attribute

Reply:
Done